package.json

```diff
@@ -6,25 +6,28 @@
   "type": "module",
   "scripts": {
     "build:client": "cd src/client && vite build",
-    "deploy": "npm run build:client && devvit upload",
-    "dev": "concurrently -p \"[{name}]\" -n \"VITE,DEVVIT,GAME\" -c \"blue,green,magenta\" \"npm run dev:vite\" \"npm run dev:devvit\" \"npm run dev:client\" --restart-tries 2",
+    "build:server": "cd src/server && node esbuild.js",
+    "deploy": "npm run build:client && npm run build:server && devvit upload",
+    "dev": "concurrently -p \"[{name}]\" -n \"CLIENT,SERVER,DEVVIT\" -c \"blue,green,magenta\" \"npm run dev:client\" \"npm run dev:server\" \"npm run dev:devvit\" --restart-tries 2",
     "dev:client": "cd src/client && vite build --watch",
+    "dev:server": "cd src/server && node esbuild.js --watch",
     "dev:devvit": "devvit playtest YOUR_SUBREDDIT_NAME",
-    "dev:vite": "cd src/client && vite --port 7474",
     "login": "devvit login",
     "type-check": "tsc --build"
   },
   "dependencies": {
-    "@devvit/client": "0.11.16",
-    "@devvit/public-api": "0.11.16",
-    "@devvit/server": "0.11.16",
-    "@devvit/redis": "0.11.16",
-    "devvit": "latest",
+    "@devvit/client": "0.11.17-next-2025-06-05-be427a4fb.0",
+    "@devvit/public-api": "0.11.17-next-2025-06-05-be427a4fb.0",
+    "@devvit/server": "0.11.17-next-2025-06-05-be427a4fb.0",
+    "@devvit/redis": "0.11.17-next-2025-06-05-be427a4fb.0",
+    "devvit": "0.11.17-next-2025-06-05-be427a4fb.0",
     "express": "5.1.0"
   },
   "devDependencies": {
     "@types/express": "5.0.1",
     "concurrently": "9.1.2",
+    "esbuild": "0.20.2",
```

Comment:
consider sticking to one build tool. all esbuild or all vite. mastering two is hard and increases our dependency footprint.